### ISSUE TITLE
A.C.I.D. Vest broken

### ISSUE THREAD

**Issue by nuklearcellphoneg** [cmss13-devs/cmss13#2082]
### Testmerges

N/A

### Description of the bug

I set the vest to inject 1u at 1 damage,
Tried both or and and,
 set to burning on and off
  nominal. emergency, etc on and off
   then inserted at first a vial with 5u kelotane,
    then a bottle of 5u dermaline,
     both of them mixed,
      and also a vial with 1u dermaline
       then 1u kelotane.
       
       
       
       The vest never injected anything. It wouldn't inject water as well. Not sure what it can inject at the moment.

### What's the difference with what should have happened?

Should have injected the kelo, the derma, AND both the kelo and derma

### How do we reproduce this bug?

1. wear ACID vest
2. set it to minimum inject/damage to avoid OD detection preventing injection which is probably separate
3. See what reagents, if any, the ACID will successfully inject.

### Issue Bingo

- [X] Issue could be reproduced at least once
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

**Comment by CocaColaTastesGood** [comment]
It needs a cell inside of it, and the liquid needs to be in a vial. Worked for me, but it processes very slowly so you won't be getting instant responses. No signals or anything.

**Comment by nuklearcellphoneg** [comment]
Yeah, I used it with vials, cells, different cells, different size beakers, et al-

Never injects. S'why I tried it with just kelotane and dermaline, with amounts that were less than their OD limits- it just never ever injects. Now it may inject some medicines, sure- but

Let's say you use a vial with 10/10/10 kelo/derma/oxy, and it's meant to inject 5u once any damage is done- it just never will. I ran into a similar problem ages ago, but the vest notified me in chat why it wouldn't inject at the time- there was 15/15 keloderma in the vial, set to inject 15u or 7.5/7.5u Keloderma- it informed me it wouldn't inject, because Derma is an OD at 15u- to save on it costing the server to check reagent amounts or whatever, all the vest checks is a. reagent types b. is the injection amount bigger any of those reagents' OD amount- if so don't inject.

However, that's not what seems to be happening now and I'm unsure of what's causing it in this instance.

**Comment by Huffie56** [comment]
i tested the vest and it seem to work with bicardine.... to much complicated for me to set up ....

**Comment by nuklearcellphoneg** [comment]
For some reason, at the time I could not get the vest to inject any amount of kelotane or dermaline at any amount.

**Comment by Huffie56** [comment]
> For some reason, at the time I could not get the vest to inject any amount of kelotane or dermaline

so would you say this issue is solved? because from what i understand this issue complain about the vest not working at all.
so i think it's solve ? you can close it if you agree and if you have other issue with the vest open a new issue. thanks

**Comment by nuklearcellphoneg** [comment]
i'll try to retest tonight

**Comment by nuklearcellphoneg** [comment]
Seems to work fine now; could still use an option to disable the safeties methinks, and I think that if you remove the battery and put it back in, this should cause it to rescan + eject-

Not sure how long it takes exactly between injections, but long enough that I tried everything I could to force an injection and was very close to keeping this open, as I almost assumed it was still broken as it took. . . maybe five minutes before it would inject again?

**Comment by Huffie56** [comment]
> Seems to work fine now; could still use an option to disable the safeties methinks, and I think that if you remove the battery and put it back in, this should cause it to rescan + eject-
> 
> Not sure how long it takes exactly between injections, but long enough that I tried everything I could to force an injection and was very close to keeping this open, as I almost assumed it was still broken as it took. . . maybe five minutes before it would inject again?

thanks for taking the time to retest it.
and don't hesitate to post other issues.
if you have suggestion about the vest you can post them on the forum.
